### DISCUSSION TITLE
Static binary of NXC

### DISCUSSION THREAD

**Discussion by Xe1Dery** [Pennyw0rth/NetExec#127] [Q&A] [answered]
Hello, we are engaged in red teaming and we have a need for a static image of NXC. The binary file in Releases requires additional libraries to run. Is there any manual on how to make NXC static? (The docker image works but is too big)

![nxc](https://github.com/Pennyw0rth/NetExec/assets/124486670/fd943032-cab2-4422-8158-ae5f91b26243)
![nxc2](https://github.com/Pennyw0rth/NetExec/assets/124486670/31c75139-c7c8-4a5b-b581-c3f4cdf93afe)

**Comment by NeffIsBack** [comment]
We use pyinstaller together with the netexec.spec file to build the binaries. You can take a look at the workflow file at `.github/workflows/netexec-build.yml`, but I am not sure which libraries are used.

**Reply by NeffIsBack** [reply, marked as answer]
@Xe1Dery i have published a tutorial on the wiki now: https://www.netexec.wiki/getting-started/installation/manually-building-the-binaries

**Comment by andretorresbr** [comment]
Hi guys.
I followed the wiki and the tool apparently worked correctly on Windows as a single executable.
However, everytime I try to use a module, I receive this error:

`[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\adcs.py filename must match the module name adcs
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\add-computer.py filename must match the module name add-computer
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\bh_owned.py filename must match the module name bh_owned
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\daclread.py filename must match the module name daclread
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\dfscoerce.py filename must match the module name dfscoerce
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\drop-sc.py filename must match the module name drop-sc
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\empire_exec.py filename must match the module name empire_exec
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\enum_av.py filename must match the module name enum_av
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\enum_dns.py filename must match the module name enum_dns
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\enum_trusts.py filename must match the module name enum_trusts
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\find-computer.py filename must match the module name find-computer
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\firefox.py filename must match the module name firefox
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\get-desc-users.py filename must match the module name get-desc-users
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\get-network.py filename must match the module name get-network
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\get-unixUserPassword.py filename must match the module name get-unixUserPassword
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\get-userPassword.py filename must match the module name get-userPassword
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\get_netconnections.py filename must match the module name get_netconnections
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\gpp_autologin.py filename must match the module name gpp_autologin
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\gpp_password.py filename must match the module name gpp_password
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\group-mem.py filename must match the module name group-mem
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\groupmembership.py filename must match the module name groupmembership
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\handlekatz.py filename must match the module name handlekatz
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\hash_spider.py filename must match the module name hash_spider
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\iis.py filename must match the module name iis
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\impersonate.py filename must match the module name impersonate
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\install_elevated.py filename must match the module name install_elevated
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\ioxidresolver.py filename must match the module name ioxidresolver
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\keepass_discover.py filename must match the module name keepass_discover
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\keepass_trigger.py filename must match the module name keepass_trigger
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\laps.py filename must match the module name laps
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\ldap-checker.py filename must match the module name ldap-checker
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\lsassy.py filename must match the module name lsassy
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\MAQ.py filename must match the module name MAQ
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\masky.py filename must match the module name masky
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\met_inject.py filename must match the module name met_inject
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\ms17-010.py filename must match the module name ms17-010
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\msol.py filename must match the module name msol
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\mssql_priv.py filename must match the module name mssql_priv
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\nanodump.py filename must match the module name nanodump
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\nopac.py filename must match the module name nopac
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\ntdsutil.py filename must match the module name ntdsutil
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\ntlmv1.py filename must match the module name ntlmv1
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\petitpotam.py filename must match the module name petitpotam
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\pi.py filename must match the module name pi
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\printnightmare.py filename must match the module name printnightmare
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\procdump.py filename must match the module name procdump
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\pso.py filename must match the module name pso
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\rdcman.py filename must match the module name rdcman
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\rdp.py filename must match the module name rdp
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\reg-query.py filename must match the module name reg-query
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\runasppl.py filename must match the module name runasppl
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\schtask_as.py filename must match the module name schtask_as
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\scuffy.py filename must match the module name scuffy
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\shadowcoerce.py filename must match the module name shadowcoerce
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\slinky.py filename must match the module name slinky
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\spider_plus.py filename must match the module name spider_plus
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\spooler.py filename must match the module name spooler
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\subnets.py filename must match the module name subnets
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\teams_localdb.py filename must match the module name teams_localdb
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\test_connection.py filename must match the module name test_connection
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\uac.py filename must match the module name uac
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\user-desc.py filename must match the module name user-desc
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\veeam.py filename must match the module name veeam
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\wcc.py filename must match the module name wcc
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\wdigest.py filename must match the module name wdigest
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\webdav.py filename must match the module name webdav
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\web_delivery.py filename must match the module name web_delivery
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\whoami.py filename must match the module name whoami
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\wifi.py filename must match the module name wifi
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\winscp.py filename must match the module name winscp
[-] C:\Users\ADMIN~1.SEC\AppData\Local\Temp\_MEI35202\nxc\modules\zerologon.py filename must match the module name zerologon
[10:06:21] ERROR    Module not found: lsassy                     netexec.py:189`

![2023-12-24 15_59_45-192 168 100 21 - Remote Desktop Connection](https://github.com/Pennyw0rth/NetExec/assets/84332391/4468dbc6-dcd6-40ba-9a43-4088cc054145)

Is there anything that I'm missing?

Regards.

**Reply by NeffIsBack** [reply]
This should not happen, thanks for the bug report, I will take a look it at this soon!

In the meantime you can try to reset to commit ef4f2df92cc553274a09c38fead0030d5018481e and compile the binary with that.

**Reply by andretorresbr** [reply]
Thanks a lot! Merry christmas :)

**Reply by NeffIsBack** [reply]
Fixed, Merry christmas :)

**Reply by andretorresbr** [reply]
I recompiled it and modules are not presenting that error again, what is great!
However I'm still having issues with some NTDS options. As an example:

![001](https://github.com/Pennyw0rth/NetExec/assets/84332391/07cc439c-435d-41b6-95fb-c07743d717c3)


Do you think I should open new issues for them?

**Reply by NeffIsBack** [reply]
That is a weird exception lol. Yes please do so
